community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledConversionUtils.java

```diff
@@ -765,7 +765,9 @@ public Object next()
             {
                 throw new NoSuchElementException();
             }
-            return Array.get( array, position++ );
+            Object result = Array.get( array, position );
+            position++;
+            return result;
```

Comment:
In what way is this an improvement?

Reply:
Let me show you what sonarQube has to say about this one:

"The use of increment and decrement operators in method calls or in combination with other arithmetic operators is not recommended, because:

- It can significantly impair the readability of the code.
- It introduces additional side effects into a statement, with the potential for undefined behavior.
- It is safer to use these operators in isolation from any other arithmetic operators."

To me personally, it feels like it improves readability a (tiny) bit

Reply:
I feel the grouping of the three things (access from array, increment, return) is wrong in this new version, making it less clear than it was before. Perhaps it would be clearer written as:
```
int offset = position++;
return Array.get( array, position );
```

Reply:
But wouldn't that access the wrong position (+1), thus we would need to access first and then increment?